json/schemas/partition-table-schema.json

```diff
@@ -0,0 +1,175 @@
+                            "type": "string",
+                            "pattern": "^(r|w){0,2}$"
+                        },
+                        "nonsecure": {
+                            "description": "Non-Secure Permissions",
+                            "type": "string",
+                            "pattern": "^(r|w){0,2}$"
+                        },
+                        "bootloader": {
+                            "description": "Bootloader Permissions",
+                            "type": "string",
+                            "pattern": "^(r|w){0,2}$"
+                        }
+                    }
+                }
+            },
+            "required": ["permissions", "families"],
+            "additionalProperties": false
+        },
+        "partitions": {
+            "description": "Partitions",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "description": "Partition Name",
+                        "type": "string"
+                    },
+                    "id": {
+                        "description": "Partition ID",
+                        "type": ["integer", "string"],
+                        "minimum": 0,
+                        "exclusiveMaximum": 18446744073709551616,
+                        "pattern": "^0x[0-9a-fA-F]{1,16}$",
+                        "examples": [
+                            "0xDED3FFFF01234567",
+                            29,
+                            "0xdeadbeef"
+                        ]
+                    },
+                    "start": {
+                        "description": "Partition Start",
+                        "type": ["integer", "string"],
+                        "minimum": 0,
+                        "pattern": "^\\d+(k|K)$"
+                    },
+                    "size": {
+                        "description": "Partition Size",
+                        "type": ["integer", "string"],
+                        "minimum": 0,
+                        "pattern": "^\\d+(k|K)$"
+                    },
+                    "families": {
+                        "description": "UF2 families accepted",
+                        "type": "array",
+                        "items": {
+                            "type": "string",
+                            "pattern": "^data|absolute|rp2040|rp2350-arm-s|rp2350-arm-ns|rp2350-riscv|0x[0-9a-fA-F]{1,8}$",
+                            "examples": [
+                                "data",
+                                "absolute",
+                                "rp2040",
+                                "rp2350-arm-s",
+                                "rp2350-arm-ns",
+                                "rp2350-riscv"
+                            ]
+                        }
+                    },
+                    "permissions": {
```

Comment:
It looks like this `permissions` part of the schema is identical to the earlier `permissions` part? Might be worth using https://json-schema.org/understanding-json-schema/structuring#defs to avoid repetition.